docker/docker.go

```diff
@@ -28,8 +28,47 @@ import (
 
 type Port = types.Port
 
+var dockerClient *client.Client
```

Comment:
is it possible that we will call `GetClient` concurrently?

Reply:
mhmmm... that's a very good point. Honestly, I don't know. But just in case I think that it is expected `GetClient` to be thread-safe. I'd add a mutex when creating the client. WDYT?

Reply:
Why do you need singleton here at all? Because of the overhead? How big is it?

Reply:
> Why do you need singleton here at all? Because of the overhead?

Mainly because I found a bit awkward to build the client every time.

> How big is it?

I haven't actually measured it.

Reply:
I've tried [here](https://github.com/src-d/engine/pull/428/commits/41ca684153749fc32cf8e74412d8df4ce0a9e582) to make it thread-safe. Let me know wdyt and if you prefer each function to build its own client.

Reply:
I'm also not sure if a singleton is needed.
I could call `client.Close()` somewhere and break the client of other goroutine, for example.

Reply:
True. I guess that it's not that safe in this case to make a singleton without actually wrapping the original client to avoid for example exposing the `Close` method or others that might break. But at least for now it seems an unneeded overkill given that the overhead shouldn't be an issue.

Reply:
I see neither the necessity of making the client a singleton.
In which circumstances do we need to ensure there is only one instance of the docker client?

Reply:
> In which circumstances do we need to ensure there is only one instance of the docker client?

No actually need to ensure uniqueness, I just found a bit awkward to create a client every time we need it. But since it introduces more issues than it "solves" I'm gonna remove it.

Reply:
Removed singleton [here](https://github.com/src-d/engine/pull/428/commits/71c5d42bc85eccb016401709ca5cac7fae7e63e4).